src/psyclone/psyir/transformations/hoist_local_arrays_trans.py

```diff
@@ -226,27 +226,25 @@ def apply(self, node, options=None):
                                     cond_expr, expr)
                     check_added = True
 
-            # TODO #1366: we also have to use a CodeBlock for the allocate()
-            # and deallocate() inside the conditional body.
             body = []
             if check_added:
                 body.append(
                     IfBlock.create(
                         allocated_expr.copy(),
-                        [freader.psyir_from_statement(
-                            f"deallocate({sym.name})",
-                            node.symbol_table)]))
+                        [IntrinsicCall.create(
+                            IntrinsicCall.Intrinsic.DEALLOCATE,
+                            [Reference(sym)])]))
             body.append(
-                freader.psyir_from_statement(f"allocate({fwriter(aref)})",
```

Comment:
Also remove the `fwriter = FortranWriter()` at line 157 and amend the comment.

Reply:
Oops, good spot.